mirrord/intproxy/src/agent_conn.rs

```diff
@@ -157,3 +222,54 @@ impl BackgroundTask for AgentConnection {
         }
     }
 }
+
+pub async fn wrap_connection_with_tls(
+    stream: TcpStream,
+    proxy_addr: IpAddr,
+    tls_certificate: &Path,
+    client_tls_certificate: &Path,
+    client_tls_key: &Path,
+) -> Result<(mpsc::Sender<ClientMessage>, mpsc::Receiver<DaemonMessage>), ConnectionTlsError> {
+    let mut root_cert_store = rustls::RootCertStore::empty();
+
+    root_cert_store.add_parsable_certificates(
+        rustls_pemfile::certs(&mut BufReader::new(File::open(tls_certificate).map_err(
```

Comment:
I think this is the second time I'm seeing this stuff? Can't you put in some shared place and turn it into some extension trait (if that's true)?

Reply:
I mean the whole tls cert setup.

Reply:
I think of making mirrord-tls for all tls stuff including the errors and agent tls connection

Reply:
Well I've removed it from the other one because CLI should never connect to external-proxy but a thing intproxy does